src/components/postComponent/components/post-geolocation.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+	import { getToastStore } from '@skeletonlabs/skeleton';
+	import Geolocation from 'svelte-geolocation';
+
+	export let coords: [number, number];
+	let getPosition = false;
+	let isLocationActivated = false;
+	let buttonDisabled = false;
+
+	const toastStore = getToastStore();
+	const buttonLoadingText = 'Geolocation wird geladen';
+	const buttonActivationText = 'Geolocation Aktivieren';
+	const buttonActivatedText = 'Geolocation aktiviert';
+	let buttonGeolocationText = buttonActivationText;
+
+	// State management for button
+	function updateButtonState(status: 'loading' | 'activated' | 'deactivated'): string {
+		switch (status) {
+			case 'loading':
+				buttonGeolocationText = buttonLoadingText;
+				buttonDisabled = true;
+				return '';
+			case 'activated':
+				buttonGeolocationText = buttonActivatedText;
+				buttonDisabled = false;
+				isLocationActivated = true;
+				return '';
+			case 'deactivated':
+				buttonGeolocationText = buttonActivationText;
+				buttonDisabled = false;
+				isLocationActivated = false;
+				coords = [-1, -1];
+				return '';
+		}
+	}
```

Comment:
Das hier wäre cleaner wenn man das mit [Svelte Statements ](https://learn.svelte.dev/tutorial/reactive-statements)macht.

Hier noch ein anderes Beispiel aus Johannas Code: 
![image](https://github.com/wwi21seb-projekt/Client_BubbleMatee/assets/93786770/773873ef-f502-4cf7-9db6-ea003905e988)

Findest du in der File `profile-header-bottom.svelte`